### DISCUSSION TITLE
seleniumbase can't find or click element, even though it exists

### DISCUSSION THREAD

**Discussion by bobberystudent** [seleniumbase/SeleniumBase#2786] [General]
hello all, the element is a checkbox near end for terms&conditions. not in an iframe either. when control F in inspect element, it can find it, but not via seleniumbase code.  It can get everything else but not this. selector stried: input#terms_condition,  '//*[@id="terms_condition"]', input.terms_conditionrnb_checkbox.rnb-opacity-0.rnb-absolute.rnb-inset-0, input[name="terms_condition

code:
```
with SB(uc=True) as sb:
      geturl()
      click_other_elements()
      sb.driver.sleep(10)
      sb.driver.uc_click("input#terms_condition")
```

![asdfsdfsdfs](https://github.com/seleniumbase/SeleniumBase/assets/169928117/02a213a3-79a7-4c59-945f-8676e47c71c9)

**Comment by mdmintz** [comment]
It looks like you forgot to include the line of code that navigates to the URL you want. Be sure to write complete scripts.

**Reply by bobberystudent** [reply]
I do have the function that gets the url, I just did not put it on the post. Seleniumbase can click all other elements I want, just not that specfic one. Is there a potential solution? Any help would be appericated, thank you

**Reply by mdmintz** [reply]
Make sure to use the correct selector. If there’s more than one element matching a selector, then use a better selector. If there’s an iframe, make sure to switch into the correct iframe first. If there are multiple tabs open, make sure you’ve switched into the correct tab first.

**Reply by bobberystudent** [reply]
sorry for bothering again, but I have done all those things above. there's only one matching selector which is the correct one, there's no iframe, and there is only one tab, yet seleniumbase still does not detect it, yet inspect element can. here is the full code I ran below. (targeting terms and condition checkbox)


from seleniumbase import SB
with SB(uc=True) as sb:
    sb.driver.uc_open_with_reconnect("https://www.routenote.com/rn/signup_page/design", 5)
    potentia_selectors = [
    ('//*[@id="terms_condition"]', "xpath"),
    ("input.terms_conditionrnb_checkbox.rnb-opacity-0.rnb-absolute.rnb-inset-0", "css selector"),
    ("input#terms_condition", "css selector"),
    ('input[name="terms_condition"]', "css selector"),
    ("(//input[@class='rnb_checkbox rnb-opacity-0 rnb-absolute rnb-inset-0'])[2]", "xpath")
    ]

    for i, (selector, by) in enumerate(potentia_selectors):
        try:
            sb.driver.uc_click(selector, by=by)
            print(f"{i} worked", "-"*150)
        except:
            pass

**Reply by mdmintz** [reply]
Try using the Recorder to generate your script: https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/recorder_mode.md

**Reply by subh-sk** [reply]
Hii @bobberystudent 
I have reviewed your code and made some adjustments to ensure it works perfectly and clicks all the elements. Here is the revised code:

```py

from seleniumbase import SB,BaseCase
def Fix_func(sb:BaseCase):
    sb.driver.uc_open_with_reconnect("https://www.routenote.com/rn/signup_page/design", 5)
    


    checkbox = sb.find_elements(selector="//input[@type='checkbox']", by="xpath")
    checkbox[0].click()
    checkbox[1].click()


    sb.switch_to_frame("#html_element > div > div > iframe")
    sb.driver.uc_click("#recaptcha-anchor > div.recaptcha-checkbox-checkmark")
    sb.driver.reconnect(timeout=2)
    sb.switch_to_parent_frame()

                    ##CODE
    input("wait = ")

with SB(uc=True) as sb:
    Fix_func(sb)
    
   ```
   
## Thank you for using the `seleniumbase` package. I hope this helps you achieve more efficient web automation.
